lib/select.c

```diff
@@ -86,18 +86,18 @@ int Curl_wait_ms(timediff_t timeout_ms)
   }
 #if defined(MSDOS)
   delay(timeout_ms);
-#elif defined(USE_WINSOCK)
+#elif defined(WIN32)
   /* prevent overflow, timeout_ms is typecast to ULONG/DWORD. */
 #if TIMEDIFF_T_MAX > ULONG_MAX
   if(timeout_ms > ULONG_MAX)
-#if ULONG_MAX < INFINITE
     timeout_ms = ULONG_MAX;
-#else
-    timeout_ms = ULONG_MAX-1;
-    /* avoid waiting forever */
-#endif
 #endif
+  /* avoid INFINITE, negated check to not care about > or >= due to max */
+#if INFINITE > 0
+  if(!(timeout_ms < INFINITE))
+    timeout_ms = INFINITE-1;
```

Comment:
IMO this is unnecessary as well, and harder to understand than
~~~
#if ULONG_MAX == INFINITE
timeout_ms = ULONG_MAX - 1;
~~~
which I don't think is even necessary because we know on Windows ULONG_MAX is always 0xFFFFFFFF and apparently so is INFINITE so why not just do it ULONG_MAX - 1

Reply:
Thanks, agreed due to https://docs.microsoft.com/en-us/cpp/cpp/integer-limits.